src/test/lib/storage/index/partial_hash/partial_hash_index_test.cpp

```diff
@@ -68,90 +95,130 @@ TEST_F(PartialHashIndexTest, IndexCoverage) {
+
+  EXPECT_EQ(index_map->at("inbox").size(), 2);
+  EXPECT_EQ(index_map->at("inbox")[0], (RowID{ChunkID{0}, ChunkOffset{7}}));
+  EXPECT_EQ(index_map->at("inbox")[1], (RowID{ChunkID{1}, ChunkOffset{7}}));
+
+  EXPECT_EQ(index_map->at("hello").size(), 1);
+  EXPECT_EQ(index_map->at("hello")[0], (RowID{ChunkID{1}, ChunkOffset{0}}));
+
+  EXPECT_EQ(index_map->at("funny").size(), 1);
+  EXPECT_EQ(index_map->at("funny")[0], (RowID{ChunkID{1}, ChunkOffset{2}}));
+
+  EXPECT_EQ(index_map->at("names").size(), 1);
+  EXPECT_EQ(index_map->at("names")[0], (RowID{ChunkID{1}, ChunkOffset{3}}));
+
+  EXPECT_EQ(index_map->at("paper").size(), 1);
+  EXPECT_EQ(index_map->at("paper")[0], (RowID{ChunkID{1}, ChunkOffset{5}}));
+
+  EXPECT_EQ(index_map->at("clock").size(), 1);
+  EXPECT_EQ(index_map->at("clock")[0], (RowID{ChunkID{1}, ChunkOffset{6}}));
 }
 
 TEST_F(PartialHashIndexTest, Iterators) {
-  auto begin = index->cbegin();
-  auto end = index->cend();
+  auto begin = cbegin(index);
+  auto end = cend(index);
 
   auto begin_copy = begin;
   EXPECT_EQ(begin_copy, begin);
   ++begin_copy;
   EXPECT_NE(begin_copy, begin);
 
-  EXPECT_EQ(std::distance(begin, end), 14);  // Test size of index iterator.
-  EXPECT_EQ(std::distance(index->null_cbegin(), index->null_cend()), 2);  // Test size of NULL values index iterator.
-  EXPECT_NE(std::find(begin, end, RowID{ChunkID{0}, ChunkOffset{4}}), end);  // Test for not-existing value in iterator.
+  // Test size of index iterator.
+  EXPECT_EQ(std::distance(begin, end), 14);
+  // Test size of NULL values index iterator.
+  EXPECT_EQ(std::distance(null_cbegin(index), null_cend(index)), 2);
+  // Test for not-existing value in iterator.
+  EXPECT_NE(std::find(begin, end, RowID{ChunkID{0}, ChunkOffset{4}}), end);
+}
+
+TEST_F(PartialHashIndexTest, AccessWithIterators) {
+  auto test_access_values_with_iterators = [](auto begin, auto end) {
+    auto current_iter = begin;
+    EXPECT_EQ(current_iter, begin);
+    ++current_iter;
+    EXPECT_NE(current_iter, begin);
+    // Test size of index iterator.
+    EXPECT_EQ(std::distance(begin, end), 14);
+    // Test for not-existing value in the iterator.
+    EXPECT_NE(std::find(begin, end, RowID{ChunkID{0}, ChunkOffset{4}}), end);
+  };
+  index->access_values_with_iterators(test_access_values_with_iterators);
+
+  auto access_null_values_with_iterators = [](auto begin, auto end) {
+    // Test size of NULL values index iterator.
+    EXPECT_EQ(std::distance(begin, end), 2);
+  };
+  index->access_null_values_with_iterators(access_null_values_with_iterators);
+
+  auto access_range_equals_with_iterators = [](auto begin, auto end) {
+    EXPECT_EQ(std::distance(begin, end), 3);
+
+    EXPECT_EQ(*begin, (RowID{ChunkID{0}, ChunkOffset{1}}));
+    ++begin;
+    EXPECT_EQ(*begin, (RowID{ChunkID{0}, ChunkOffset{3}}));
+    ++begin;
+    EXPECT_EQ(*begin, (RowID{ChunkID{1}, ChunkOffset{1}}));
+  };
+  index->range_equals_with_iterators(access_range_equals_with_iterators, "delta");
```

Comment:
Please add a test for both lookup functions with a search key for which no index entries exist.